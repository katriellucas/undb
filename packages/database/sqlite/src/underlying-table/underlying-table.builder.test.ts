import { EntityManager } from '@mikro-orm/better-sqlite'
import {
  createTestTable,
  FieldId,
  FieldName,
  FieldValueConstraints,
  StringField,
  TableSchema,
  WithTableSchema,
} from '@undb/core'
import { UnderlyingTableBuilder } from './underlying-table.builder.js'

describe('UnderlyingTableBuilder', () => {
  let em: EntityManager

  beforeAll(() => {
    // @ts-expect-error
    em = global.em
  })

  test('should build table', () => {
    const queries = new UnderlyingTableBuilder(em)
      .createTable(
        createTestTable(
          new WithTableSchema(
            new TableSchema([
              new StringField({
                id: FieldId.fromString('fldid'),
                name: FieldName.create('name'),
                valueConstrains: FieldValueConstraints.create({}),
              }),
            ]),
          ),
        ),
      )
      .build()

    expect(queries).toMatchSnapshot()
  })
})
