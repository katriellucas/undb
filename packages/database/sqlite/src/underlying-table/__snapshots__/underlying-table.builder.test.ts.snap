// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnderlyingTableBuilder > should build table 1`] = `
[
  "create table if not exists \`tableId\` (\`auto_increment\` integer not null primary key autoincrement, \`id\` varchar(255) not null, \`created_at\` datetime not null default CURRENT_TIMESTAMP, \`created_by\` varchar(255) not null, \`updated_at\` datetime not null default CURRENT_TIMESTAMP, \`updated_by\` varchar(255) not null, \`deleted_at\` datetime null, \`deleted_by\` varchar(255), \`fldid\` varchar(255))",
  "
      create unique index IF NOT EXISTS \`tableId_id_unique\` on \`tableId\` (\`id\`)
      ",
  "
		CREATE TRIGGER IF NOT EXISTS update_at_update_tableId AFTER UPDATE ON \`tableId\`
		BEGIN
			update \`tableId\` SET updated_at = datetime('now') WHERE id = NEW.id;
		END;
	 ",
]
`;
